packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java

```diff
@@ -96,11 +96,12 @@ public Element render() {
         for (Object semanticElement : semanticElements) {
             VariableManager nodeVariableManager = variableManager.createChild();
             nodeVariableManager.put(VariableManager.SELF, semanticElement);
+            nodeVariableManager.put(IDiagramEvent.DIAGRAM_EVENT, optionalDiagramEvent.orElse(null));
```

Comment:
We could probably make this variable available to everybody in the rendering (DiagramComponent, EdgeComponent, etc)

Reply:
Done